cmd/snap-confine/snap-confine-invocation.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "snap-confine-invocation.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "../libsnap-confine-private/cleanup-funcs.h"
+#include "../libsnap-confine-private/snap.h"
+#include "../libsnap-confine-private/string-utils.h"
+#include "../libsnap-confine-private/utils.h"
+
+void sc_init_invocation(sc_invocation *inv, const struct sc_args *args, const char *snap_instance) {
+    /* Snap instance name is conveyed via untrusted environment. It may be
+     * unset (typically when experimenting with snap-confine by hand). It
+     * must also be a valid snap instance name. */
+    if (snap_instance == NULL) {
+        die("SNAP_INSTANCE_NAME is not set");
```

Comment:
This feels like the error message is at the wrong place now, I mean: `sc_init_invocation` has no idea where snap_instance comes from. 

I think we should die() here but with a error message local to the function (like "cannot call sc_init_invocation with snap_instance == NULL" or something) and also check for NULL in the getenv("SNAP_INSTANCE_NAME") and die there with the old error message (and maybe cleaup the error message to say "environemtn SNAP_INSTANCE_NAME is not set" or something like this.

Alternatively call getenv() inside sc_init_invocation() and not pass as parameter.

Reply:
Since this error message is the only thing displayed when snap-confine is invoked without setting the environment variable I would rather ensure the message is useful. We can move the exact same check to `snap-confine.c`'s main function and just ensure that it is not NULL here.